.github/workflows/main.yml

```diff
@@ -0,0 +1,50 @@
+name: CI
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.10']
```

Comment:
I would suggest to use python 3.9 or lower that doesn't support the grammar `int | None` to better test the type hint.

Reply:
Good idea!